### DISCUSSION TITLE
Permissions on RelationManager

### DISCUSSION THREAD

**Discussion by Wulfheart** [filamentphp/filament#5471] [Help] [answered]
I would like to make the ability to create records in a relation manager conditional based on some data of the parent record. Is there any way to ensure this? i don't think that `disabledOn` and `visible` work here as they could also be called by manipulating the site's content, couldn't they? Furthermore it doesn't work with the opened modal which does not close on a specific condition (e.g. that it is not allowed to create more resource because some permission has changed).

### Example

![image](https://user-images.githubusercontent.com/25671390/212480948-50a4c26c-ef9b-4933-96c1-3615c01e5623.png)
````php
class SignedUpUsersRelationManager extends RelationManager
{
    protected static string $relationship = 'signedUpUsers';

    protected static ?string $recordTitleAttribute = 'name';

    public static function getModelLabel(): string
    {
        return __('core/game_signed_up_user.name_singular');
    }

    public static function getPluralModelLabel(): string
    {
        return __('core/game_signed_up_user.name_plural');
    }


    public static function form(Form $form): Form
    {
        return $form
            ->schema(function (RelationManager $livewire) {
                /** @var Game $game */
                $game = $livewire->getOwnerRecord();

                return [
                    Forms\Components\Select::make('user_id')
                        ->label(__('users/user.name_singular'))
                        ->options(function () use ($game) {
                            return User::query()
                                ->whereNotSignedUpForGame($game->id)
                                ->whereNotPlayingInGame($game->id)
                                ->pluck('name', 'id');
                        })
                        ->searchable()
                        ->required(),
                ];
            });
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('user.name')
                    ->label(__('users/user.name_singular'))
                    //->url(fn (Model $record) => 'https://zeit.de')
                    ->openUrlInNewTab(),
            ])
            ->filters([

            ])
            ->headerActions([
                Tables\Actions\CreateAction::make(), // <-- Here I would add the 
            ])
            ->actions([
                Tables\Actions\DeleteAction::make()->label(__('core/game_signed_up_user.actions.remove')),
            ])
            ->bulkActions([
                //Tables\Actions\DeleteBulkAction::make(),
            ]);
    }
}
````

**Comment by danharrin** [comment, marked as answer]
`Tables\Actions\CreateAction::make()->hidden(fn ($livewire) => $livewire->ownerRecord->...)`

**Reply by Wulfheart** [reply]
Is then also the action not executable if I modify the page content? E.g. could someone manually somehow bypass the security measures and execute the action even though it is not visible or is hidden?

**Reply by danharrin** [reply]
If the action is hidden, we have protection

**Reply by Wulfheart** [reply]
Ah. Nice. Didn’t know that!

**Reply by danharrin** [reply]
It also has protection if it's `disabled()`

**Reply by Wulfheart** [reply]
Just a UX note: It would be somehow helpful to close the dialog also when the related action is hidden (with a note) or a show a note on the modal. Is this something you are actively working on or something you would accept a PR for? 
![ezgif-2-a38ed35c62](https://user-images.githubusercontent.com/25671390/212542127-3505ad03-78a6-49bb-a670-73e59557477f.gif)

**Reply by danharrin** [reply]
We're not working on it, you can PR if you want

**Reply by Wulfheart** [reply]
Thanks for the clarification.
